### PR TITLE
upload multiple files concurrently new exported function

### PR DESCRIPTION
**Pull request by jdelbick**
## Description
new api for multiple files:
```
await uploadFilesConcurrently([{
                filepath: 'test-transfer-file-1.dat', 
                target: {
                    urls: [
                        'http://test-status-201/path/to/file-1-1.ext',
                        'http://test-status-201/path/to/file-1-2.ext'
                    ],
                    maxPartSize: 8
                }
            }, {
                filepath: 'test-transfer-file-2.dat', 
                target: {
                    urls: [
                        'http://test-status-201/path/to/file-2-1.ext',
                        'http://test-status-201/path/to/file-2-2.ext'
                    ],
                    maxPartSize: 8
                }
            }], options);
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/file.js [hunk 1]
```diff
@@ -276,6 +276,82 @@ async function uploadMultiPartFileConcurrently(filepath, target, options = {}) {
...
+async function uploadFilesConcurrently(files, options = {}) {
+    const uploadFiles = [];
+    for (const file of files) {
+        const target = file.target;
+        const filepath = file.filepath;
+        // fail fast, keep error handling the same as older, non-concurrent function
+        if (!target) {
+            throw new IllegalArgumentError('target not provided', target);
+        } else if (!target.urls || target.urls.length === 0) {
+            throw new IllegalArgumentError('\'target.urls\' but be a non-empty array', target.urls);
```
Comment (tmathern) [inline lib/file.js:321]:
```suggestion
            throw new IllegalArgumentError('\'target.urls\' must be a non-empty array', target.urls);
```

File: lib/file.js [hunk 2]
```diff
@@ -276,6 +276,82 @@ async function uploadMultiPartFileConcurrently(filepath, target, options = {}) {
...
+    options.preferredPartSize = options.partSize ? options.partSize : options.preferredPartSize;
+    
+    // extract upload options
+    const uploadOptions = filterObject(
+        options || {},
+        ['method', 'headers', ...CONCURRENCY_OPTIONS_FILTER,
+            ...RETRY_OPTIONS_FILTER, 'preferredPartSize']
+    );
+    const uploader = new BlockUpload();
+    console.log('UPLOAD FILES', uploadFiles)
```
Comment (tmathern) [inline lib/file.js:348]: Can be dropped once debugging is done

File: test/file-concurrently.test.js [hunk 3]
```diff
@@ -0,0 +1,2478 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
```
Comment (tmathern) [inline test/file-concurrently.test.js:2]:
```suggestion
 * Copyright 2022 Adobe. All rights reserved.
```

**Review by tmathern — APPROVED** [review]
May need documentation updates

File: test/file-concurrently.test.js (hunk 3 above)
Comment (jdelbick) [reply test/file-concurrently.test.js:2]: yikes, its already 2022!

**Pull request merged**